### DISCUSSION TITLE
useId hook implementation

### DISCUSSION THREAD

**Discussion by rChaoz** [mantinedev/mantine#2931] [Q&A]
The useId hook, if `React.useId()` is not available, is implemented like so:
```ts
function useClientId() {
  const [uuid, setUuid] = useState('');
  
    useIsomorphicEffect(() => {
      setUuid(randomId());
    }, []);
  
    return uuid;
}
```

This causes the hook to initially return an empty string on the initial render, as well as immediately causing a second re-render. My question is, why not return the correct ID on the first render, using a ref:
```ts
function useClientId() {
  return useRef(randomId()).current;
}
```
Or even a memo:
```ts
function useClientId() {
  return useMemo(randomId, []);
}
```

These 2 approaches work for me (with client-side rendering). Is there an issue with this approach with SSR?

**Comment by rtivital** [comment]
It will result in hydration mismatch because client and server will have different random id.
